services/motion/builtin/builtin.go

```diff
@@ -237,7 +258,104 @@ func (ms *builtIn) MoveOnGlobe(
+	localizer, ok := ms.localizer[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	// assert localizer as movementSensor to get current geo point
+	movementSensor, ok := localizer.(movementsensor.MovementSensor)
+	if !ok {
+		return false, fmt.Errorf("cannot assert localizer of type %T as movementSensor", localizer)
+	}
+
+	currentGP, _, _ := movementSensor.Position(ctx, nil)
+	straightlineDistance := currentGP.GreatCircleDistance(destination)
+
+	// convert destination into spatialmath.Pose with respect to lat = 0 = lng
+	dstPose := spatialmath.GeoPointToPose(destination)
+	dstPIF := referenceframe.NewPoseInFrame(referenceframe.World, dstPose)
+
+	// convert GeoObstacles into GeometriesInFrame then into a Worldstate
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles)
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	wrldst, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return false, err
+	}
+
+	// construct limits
+	limits := []referenceframe.Limit{
+		{Min: -straightlineDistance * 3, Max: straightlineDistance * 3},
+		{Min: -straightlineDistance * 3, Max: straightlineDistance * 3},
+		{Min: -2 * math.Pi, Max: 2 * math.Pi},
+	}
+
+	// create a KinematicBase from the componentName
+	baseComponent, ok := ms.components[componentName]
+	if !ok {
+		return false, fmt.Errorf("only Base components are supported for MoveOnGlobe: could not find an Base named %v", componentName)
+	}
+	kw, ok := baseComponent.(base.KinematicWrappable)
+	if !ok {
+		return false, fmt.Errorf("cannot move base of type %T because it is not KinematicWrappable", baseComponent)
+	}
+	kb, err := kw.WrapWithKinematics(ctx, localizer, limits)
+	if err != nil {
+		return false, err
+	}
+
+	// get current position
+	inputs, err := kb.CurrentInputs(ctx)
+	if err != nil {
+		return false, err
+	}
+	inputMap := map[string][]referenceframe.Input{componentName.Name: inputs}
+
+	// Add the kinematic wheeled base to the framesystem
+	if err := fs.AddFrame(kb.ModelFrame(), fs.World()); err != nil {
+		return false, err
+	}
+
+	// check for cancelled context before we are start planning
+	if ctx.Err() != nil {
+		ms.logger.Info("successfully canceled motion service MoveOnGlobe")
+		return true, ctx.Err()
+	}
+
+	// make call to motionplan
+	plan, err := motionplan.PlanMotion(ctx, ms.logger, dstPIF, kb.ModelFrame(), inputMap, fs, wrldst, nil, extra)
+	if err != nil {
+		return false, err
+	}
+
+	// check for cancelled context after we are done planning
+	if ctx.Err() != nil {
+		ms.logger.Info("successfully canceled motion service MoveOnGlobe")
+		return true, ctx.Err()
+	}
+
+	// execute the plan
+	for _, step := range plan {
+		for _, inputs := range step {
+			if len(inputs) == 0 {
+				continue
+			}
+			if err := kb.GoToInputs(ctx, inputs); err != nil {
+				// if context is cancelled we only stop the base
+				kb.Stop(ctx, nil)
```

Comment:
This could be useful in which case we would want to add it to the other motion methods, right?  @biotinker is this something we should think about?

Reply:
Do we not want to stop everything we are actively moving in this case?

However, in this particular case note that a context cancellation is only one example of many things that could cause GoToInputs to error.

In the case where the context is cancelled, we should break out of these `for` loops immediately. While other errors may be recoverable.

Reply:
I think we should avoid adding this for now.  Down the line we may need to do an audit of our stopping behavior and maybe its best to keep this logic simple and consistent across our methods for now